File: src/main/java/org/occidere/githubnotifier/vo/GithubRepository.java
package org.occidere.githubnotifier.vo;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

/**
 * @author occidere
 * @since 2020. 03. 31.
 * Blog: https://blog.naver.com/occidere
 * Github: https://github.com/occidere
 */
@Getter
@Setter
@Document(
        indexName = "github-repos",
        shards = 5,
        replicas = 1,
        refreshInterval = "60s",
        createIndex = true,
        type = "_doc"
)
@ToString
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
@NoArgsConstructor
public class GithubRepository {
    @Id
    private String id;
    private String name;
    private String ownerId;
    private String ownerLogin;
    private String description;
    @JsonProperty("stargazers_count")
    private int stargazersCount;
    @JsonProperty("watchers_count")
    private int watchersCount;
    @JsonProperty("forks_count")
    private int forksCount;

    private List<String> stargazersLogin = new ArrayList<>();
    private List<String> watchersLogin = new ArrayList<>();
    private List<String> forksLogin = new ArrayList<>();

    @JsonProperty("owner")
    private void unpackOwner(Map<String, Object> owner) {
        this.ownerId = String.valueOf(owner.get("id"));
        this.ownerLogin = (String) owner.get("login");
    }
}

File: src/main/scala/org/occidere/githubnotifier/vo/GithubRepository.scala
package org.occidere.githubnotifier.vo

import com.fasterxml.jackson.annotation.{JsonIgnoreProperties, JsonInclude, JsonProperty}
import lombok.ToString
import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document

import scala.collection.mutable

/**
 * @author occidere
 * @Blog: https://blog.naver.com/occidere
 * @Github: https://github.com/occidere
 * @since 2020. 08. 15.
 */
@Document(
  indexName = "github-repos",
  shards = 5,
  replicas = 1,
  refreshInterval = "60s",
  createIndex = true,
)
@ToString
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
class GithubRepository {
  @Id
  var id: String = _
  var name: String = _
  var ownerId: String = _
  var ownerLogin: String = _
  var description: String = _
  @JsonProperty("stargazers_count")
  var stargazersCount: Int = _
  @JsonProperty("watchers_count")
  var watchersCount: Int = _
  @JsonProperty("forks_count")
  var forksCount: Int = _

  // FIXME var
  var stargazersLogin: Iterable[String] = mutable.ListBuffer[String]()
  var watchersLogin: Iterable[String] = mutable.ListBuffer[String]()
  var forksLogin: Iterable[String] = mutable.ListBuffer[String]()

  @JsonProperty("ownser")
  private def unpackOwner(owner: Map[String, Any]): Unit = {
    ownerId = owner("id").toString
    ownerLogin = owner("login").toString
  }
}